eventstore/sql.go

```diff
@@ -95,64 +86,63 @@ func (s SQLStore[T]) Load(ctx context.Context, id stream.ID) (*T, error) {
 		_ = results.Close()
 	}()
 
-	var dbEvents []event
+	var (
+		streamID      string
+		streamVersion int
+		eventName     string
+		eventPayload  []byte
+		events        []esja.VersionedEvent[T]
+	)
 	for results.Next() {
-		e := event{}
-
-		err = results.Scan(&e.streamID, &e.streamVersion, &e.eventName, &e.eventPayload)
+		err = results.Scan(&streamID, &streamVersion, &eventName, &eventPayload)
 		if err != nil {
 			return nil, fmt.Errorf("error reading row result: %w", err)
 		}
 
-		dbEvents = append(dbEvents, e)
-	}
-
-	if len(dbEvents) == 0 {
-		return nil, ErrStreamNotFound
-	}
-
-	var events []stream.VersionedEvent[T]
-	for _, e := range dbEvents {
-		event, err := s.config.Mapper.New(e.eventName)
+		event, err := s.config.Mapper.New(eventName)
 		if err != nil {
 			return nil, fmt.Errorf("error creating new event instance: %w", err)
 		}
 
-		err = s.config.Marshaler.Unmarshal(e.eventPayload, event)
+		err = s.config.Marshaler.Unmarshal(eventPayload, event)
 		if err != nil {
 			return nil, fmt.Errorf("error unmarshaling event payload: %w", err)
 		}
 
-		mappedEvent, err := s.config.Mapper.FromTransport(ctx, e.streamID, event)
+		mappedEvent, err := s.config.Mapper.FromTransport(ctx, streamID, event)
```

Comment:
Is this correctly merged with `master`? If we want to leave the loop as it is now, this should probably use `id`:

```suggestion
		mappedEvent, err := s.config.Mapper.FromTransport(ctx, id, event)
```

And we can drop the `streamID` variable entirely.

Reply:
sure! I will fix all places where we have it.

Reply:
Fixed